### PR TITLE
Implement toContact() method

### PR DESCRIPTION
**Pull request by su-chang**
Implement toContact() method

## I'm submitting a...

```
[ ] Bug Fix
[ ] Feature
[ ] Other (Refactoring, Added tests, Documentation, ...)
```

## Checklist

- [ ] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
  - A feature commit message is prefixed "feat:"
  - A bugfix commit message is prefixed "fix:"
- [ ] Tests for the changes have been added

## Description

_please describe the changes that you are making_

_for features, please describe how to use the new feature_

_please include a reference to an existing issue, if applicable_

## Does this PR introduce a breaking change?

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/user/message.ts [hunk 1]
```diff
@@ -900,14 +900,19 @@ export class Message extends Accessory implements Sayable {
...
 
-    // TODO: return the ShareCard Contact
-    const contact = this.wechaty.userSelf()
+    const contactId = await this.puppet.messageContact(this.id)
+
+    if (!contactId) {
+      throw new Error(`can not get Contact id by message: ${contactId}`)
+    }
+
+    const contact = this.wechaty.Contact.load(contactId)
```
Comment (huan) [inline src/user/message.ts]: Add a `await contact.ready()` from here.

**Review by huan — CHANGES_REQUESTED** [review]
Please get approvement from @windmemory before I can continue reviewing this PR.

And also remember to link to the related issue from here.

Thanks.

**Review by windmemory — CHANGES_REQUESTED** [review]

**Comment by su-chang** [comment]
Related issue: https://github.com/Chatie/wechaty/issues/1855

**Review by huan — CHANGES_REQUESTED** [review]
Please update the `wechaty-puppet` version to 0.17 in  `package.json` so that this PR can test against to the latest version of the puppet abstract class.

See: 

https://github.com/Chatie/wechaty/blob/master/package.json#L101

**Comment by su-chang** [comment]
@huan I think I need your help, "This branch has conflicts that must be resolved" but it need `write access`. Sorry for my poor git skills.

**Pull request closed without merging**